CMakeLists.txt

```diff
@@ -41,6 +42,15 @@ else ()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
 endif ()
 
+# Verify the code conforms to ISO standards regarding C++ as
+# indicated in the GCC options
+# https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
+# Moreover, treat any compilation warning as an error.
+if ( SHOW_COMPILE_WARNINGS )
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wpedantic -Wextra -Werror")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wpedantic -Wextra -Werror")
```

Comment:
Be careful with using `-Wextra`. It can produce overly pedantic warnings.  Coupling this with `-Werror` could get annoying real fast.  Perhaps a selected subset of the warnings by enabled by `-Wextra` can be useful. Examples: `-Wunused-parameter` and `-Wunused-but-set-parameter`.  I'd recommend to also enable `-Wshadow`

Reply:
@conradsnicta will do. Thanks for the suggestion!